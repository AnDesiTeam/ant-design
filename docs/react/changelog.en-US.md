---
order: 10
title: Change Log
toc: false
---

<embed src="../../CHANGELOG.en-US.md"></embed>
