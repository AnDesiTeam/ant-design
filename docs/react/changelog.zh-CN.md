---
order: 10
title: 更新日志
toc: false
---

<embed src="../../CHANGELOG.zh-CN.md"></embed>
