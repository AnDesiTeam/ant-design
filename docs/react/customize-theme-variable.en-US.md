---
order: 7.1
title: Dynamic Theme (Experimental)
---

Besides [less customize theme](/docs/react/customize-theme), We also provide CSS Variable version to enable dynamic theme. You can check on [ConfigProvider](/components/config-provider/#components-config-provider-demo-theme) demo.

## Notice

This function need CSS Variable support which means it can not support IE. Please make sure your browser requirement.

## How to use

### Import antd.variable.min.css

Replace your import style file with CSS Variable version:

```diff
-- import 'antd/dist/antd.min.css';
++ import 'antd/dist/antd.variable.min.css';
```

Note: You need remove `babel-plugin-import` for the dynamic theme.

### Static config

Call ConfigProvider static function to modify theme color:

```ts
import { ConfigProvider } from 'antd';

ConfigProvider.config({
  theme: {
    primaryColor: '#25b864',
  },
});
```

## Conflict resolve

CSS Variable use `--ant` prefix by default. When multiple antd style file exist in your project, you can modify the prefix to fix it.

### Adjust

Modify `prefixCls` on the root of ConfigProvider:

```tsx
import { ConfigProvider } from 'antd';

export default () => (
  <ConfigProvider prefixCls="custom">
    <MyApp />
  </ConfigProvider>
);
```

You also need call the static function to modify `prefixCls`:

```ts
ConfigProvider.config({
  prefixCls: 'custom',
  theme: {
    primaryColor: '#25b864',
  },
});
```

### Compile less

Since prefix is modified. Origin `antd.variable.css` should also be replaced:

```bash
lessc --modify-var="ant-prefix=custom" antd/dist/antd.variable.less modified.css
```
