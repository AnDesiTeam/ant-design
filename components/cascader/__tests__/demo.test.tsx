import * as React from 'react';
import Cascader from '..';
import demoTest, { rootPropsTest } from '../../../tests/shared/demoTest';

demoTest('cascader', {
  testRootProps: false,
});

rootPropsTest('cascader', (props) => <Cascader {...props} />, {
  findRootElements: () => document.querySelectorAll('.ant-cascader, .ant-cascader-dropdown'),
  expectCount: 2,
});
