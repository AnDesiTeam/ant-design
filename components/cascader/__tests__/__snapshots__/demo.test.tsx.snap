// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components/cascader/demo/basic.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    >
      Please select
    </span>
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`renders components/cascader/demo/change-on-select.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    />
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`renders components/cascader/demo/custom-dropdown.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    >
      Please select
    </span>
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`renders components/cascader/demo/custom-render.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
  style="width:100%"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-item"
    >
      <span>
        Zhejiang
        <!-- -->
         /
      </span>
      <span>
        Hangzhou
        <!-- -->
         /
      </span>
      <span>
        West Lake
        <!-- -->
         (
        <a>
          752100
        </a>
        )
      </span>
    </span>
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
  <span
    aria-hidden="true"
    class="ant-select-clear"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      aria-label="close-circle"
      class="anticon anticon-close-circle"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`renders components/cascader/demo/custom-trigger.tsx correctly 1`] = `
<span>
  Unselect
  <!-- -->
  <a
    href="#"
  >
    Change city
  </a>
</span>
`;

exports[`renders components/cascader/demo/default-value.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-item"
      title="Zhejiang / Hangzhou / West Lake"
    >
      Zhejiang / Hangzhou / West Lake
    </span>
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
  <span
    aria-hidden="true"
    class="ant-select-clear"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      aria-label="close-circle"
      class="anticon anticon-close-circle"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`renders components/cascader/demo/disabled-option.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    />
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`renders components/cascader/demo/fields-name.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    >
      Please select
    </span>
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`renders components/cascader/demo/hover.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    />
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`renders components/cascader/demo/lazy.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        readonly=""
        role="combobox"
        style="opacity:0"
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    />
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`renders components/cascader/demo/multiple.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-multiple ant-select-allow-clear"
  style="width:100%"
>
  <div
    class="ant-select-selector"
  >
    <div
      class="ant-select-selection-overflow"
    >
      <div
        class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
        style="opacity:1;order:0"
      >
        <div
          class="ant-select-selection-search"
          style="width:0"
        >
          <input
            aria-autocomplete="list"
            aria-controls="undefined_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="undefined_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            readonly=""
            role="combobox"
            style="opacity:0"
            type="search"
            unselectable="on"
            value=""
          />
          <span
            aria-hidden="true"
            class="ant-select-selection-search-mirror"
          >
          </span>
        </div>
      </div>
    </div>
    <span
      class="ant-select-selection-placeholder"
    />
  </div>
</div>
`;

exports[`renders components/cascader/demo/placement.tsx correctly 1`] = `
Array [
  <div
    class="ant-radio-group ant-radio-group-outline"
  >
    <label
      class="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
    >
      <span
        class="ant-radio-button ant-radio-button-checked"
      >
        <input
          checked=""
          class="ant-radio-button-input"
          type="radio"
          value="topLeft"
        />
        <span
          class="ant-radio-button-inner"
        />
      </span>
      <span>
        topLeft
      </span>
    </label>
    <label
      class="ant-radio-button-wrapper"
    >
      <span
        class="ant-radio-button"
      >
        <input
          class="ant-radio-button-input"
          type="radio"
          value="topRight"
        />
        <span
          class="ant-radio-button-inner"
        />
      </span>
      <span>
        topRight
      </span>
    </label>
    <label
      class="ant-radio-button-wrapper"
    >
      <span
        class="ant-radio-button"
      >
        <input
          class="ant-radio-button-input"
          type="radio"
          value="bottomLeft"
        />
        <span
          class="ant-radio-button-inner"
        />
      </span>
      <span>
        bottomLeft
      </span>
    </label>
    <label
      class="ant-radio-button-wrapper"
    >
      <span
        class="ant-radio-button"
      >
        <input
          class="ant-radio-button-input"
          type="radio"
          value="bottomRight"
        />
        <span
          class="ant-radio-button-inner"
        />
      </span>
      <span>
        bottomRight
      </span>
    </label>
  </div>,
  <br />,
  <br />,
  <div
    class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      >
        Please select
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>,
]
`;

exports[`renders components/cascader/demo/render-panel.tsx correctly 1`] = `
<div
  style="padding-bottom:0;position:relative;min-width:0"
>
  <div
    class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
    style="margin:0"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      >
        Please select
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>
</div>
`;

exports[`renders components/cascader/demo/search.tsx correctly 1`] = `
<div
  class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        role="combobox"
        type="search"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    >
      Please select
    </span>
  </div>
  <span
    aria-hidden="true"
    class="ant-select-arrow"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="ant-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`renders components/cascader/demo/showCheckedStrategy.tsx correctly 1`] = `
Array [
  <div
    class="ant-select ant-cascader ant-select-multiple ant-select-allow-clear"
    style="width:100%"
  >
    <div
      class="ant-select-selector"
    >
      <div
        class="ant-select-selection-overflow"
      >
        <div
          aria-hidden="true"
          class="ant-select-selection-overflow-item ant-select-selection-overflow-item-rest"
          style="opacity:0;height:0;overflow-y:hidden;order:9007199254740991;pointer-events:none;position:absolute"
        >
          <span
            class="ant-select-selection-item"
            title="+ 2 ..."
          >
            <span
              class="ant-select-selection-item-content"
            >
              + 2 ...
            </span>
          </span>
        </div>
        <div
          class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
          style="opacity:1;order:0"
        >
          <div
            class="ant-select-selection-search"
            style="width:0"
          >
            <input
              aria-autocomplete="list"
              aria-controls="undefined_list"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="undefined_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              readonly=""
              role="combobox"
              style="opacity:0"
              type="search"
              unselectable="on"
              value=""
            />
            <span
              aria-hidden="true"
              class="ant-select-selection-search-mirror"
            >
            </span>
          </div>
        </div>
      </div>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-clear"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        aria-label="close-circle"
        class="anticon anticon-close-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="close-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </span>
    </span>
  </div>,
  <br />,
  <br />,
  <div
    class="ant-select ant-cascader ant-select-multiple ant-select-allow-clear"
    style="width:100%"
  >
    <div
      class="ant-select-selector"
    >
      <div
        class="ant-select-selection-overflow"
      >
        <div
          aria-hidden="true"
          class="ant-select-selection-overflow-item ant-select-selection-overflow-item-rest"
          style="opacity:0;height:0;overflow-y:hidden;order:9007199254740991;pointer-events:none;position:absolute"
        >
          <span
            class="ant-select-selection-item"
            title="+ 0 ..."
          >
            <span
              class="ant-select-selection-item-content"
            >
              + 0 ...
            </span>
          </span>
        </div>
        <div
          class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
          style="opacity:1;order:0"
        >
          <div
            class="ant-select-selection-search"
            style="width:0"
          >
            <input
              aria-autocomplete="list"
              aria-controls="undefined_list"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="undefined_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              readonly=""
              role="combobox"
              style="opacity:0"
              type="search"
              unselectable="on"
              value=""
            />
            <span
              aria-hidden="true"
              class="ant-select-selection-search-mirror"
            >
            </span>
          </div>
        </div>
      </div>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-clear"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        aria-label="close-circle"
        class="anticon anticon-close-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="close-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </span>
    </span>
  </div>,
]
`;

exports[`renders components/cascader/demo/size.tsx correctly 1`] = `
Array [
  <div
    class="ant-select ant-cascader ant-select-lg ant-select-single ant-select-allow-clear ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      />
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>,
  <br />,
  <br />,
  <div
    class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      />
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>,
  <br />,
  <br />,
  <div
    class="ant-select ant-cascader ant-select-sm ant-select-single ant-select-allow-clear ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      />
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>,
  <br />,
  <br />,
]
`;

exports[`renders components/cascader/demo/status.tsx correctly 1`] = `
<div
  class="ant-space ant-space-vertical"
>
  <div
    class="ant-space-item"
    style="margin-bottom:8px"
  >
    <div
      class="ant-select ant-cascader ant-select-status-error ant-select-single ant-select-allow-clear ant-select-show-arrow"
    >
      <div
        class="ant-select-selector"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-autocomplete="list"
            aria-controls="undefined_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="undefined_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            readonly=""
            role="combobox"
            style="opacity:0"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-placeholder"
        >
          Error
        </span>
      </div>
      <span
        aria-hidden="true"
        class="ant-select-arrow"
        style="user-select:none;-webkit-user-select:none"
        unselectable="on"
      >
        <span
          class="ant-select-arrow-icon"
        />
      </span>
    </div>
  </div>
  <div
    class="ant-space-item"
  >
    <div
      class="ant-select ant-cascader ant-select-status-warning ant-select-multiple ant-select-allow-clear"
    >
      <div
        class="ant-select-selector"
      >
        <div
          class="ant-select-selection-overflow"
        >
          <div
            class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
            style="opacity:1"
          >
            <div
              class="ant-select-selection-search"
              style="width:0"
            >
              <input
                aria-autocomplete="list"
                aria-controls="undefined_list"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-owns="undefined_list"
                autocomplete="off"
                class="ant-select-selection-search-input"
                readonly=""
                role="combobox"
                style="opacity:0"
                type="search"
                unselectable="on"
                value=""
              />
              <span
                aria-hidden="true"
                class="ant-select-selection-search-mirror"
              >
              </span>
            </div>
          </div>
        </div>
        <span
          class="ant-select-selection-placeholder"
        >
          Warning multiple
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders components/cascader/demo/suffix.tsx correctly 1`] = `
Array [
  <div
    class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
    suffixicon="[object Object]"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      >
        Please select
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>,
  <br />,
  <br />,
  <div
    class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
    suffixicon="[object Object]"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      >
        Please select
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>,
  <br />,
  <br />,
  <div
    class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      >
        Please select
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>,
  <br />,
  <br />,
  <div
    class="ant-select ant-cascader ant-select-single ant-select-allow-clear ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          readonly=""
          role="combobox"
          style="opacity:0"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      >
        Please select
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <span
        class="ant-select-arrow-icon"
      />
    </span>
  </div>,
]
`;
